ui/lib/apps/SlowQuery/utils/tableColumns.tsx

```diff
@@ -125,150 +79,126 @@ function successColumn(
-    key,
-    fieldName: 'process_time',
-    minWidth: 140,
-    maxWidth: 200,
-    columnActionsMode: ColumnActionsMode.clickable,
-    onRender: (rec) => (
-      <Bar textWidth={70} value={rec.process_time} capacity={capacity}>
-        {getValueFormat('s')(rec.process_time, 1)}
-      </Bar>
-    ),
-  }
-}
+////////////////////////////////////////////////
+// util methods
 
-function memoryColumn(rows?: { memory_max?: number }[]): IColumn {
-  const capacity = rows ? max(rows.map((v) => v.memory_max)) ?? 0 : 0
-  const key = 'Mem_max'
+function avgP90MaxColumn(columnNamePrefix: string, rows?: any[]): IColumn {
+  const avgFiledName = `${columnNamePrefix}_avg`.toLowerCase()
+  const p90FiledName = `${columnNamePrefix}_p90`.toLowerCase()
+  const maxFiledName = `${columnNamePrefix}_max`.toLowerCase()
+  const capacity = rows ? max(rows.map((v) => v[maxFiledName])) ?? 0 : 0
   return {
-    name: commonColumnName('memory_max'),
-    key,
-    fieldName: 'memory_max',
+    name: commonColumnName(TRANS_KEY_PREFIX, avgFiledName),
+    key: `${columnNamePrefix}_avg`,
+    fieldName: avgFiledName,
     minWidth: 140,
     maxWidth: 200,
     columnActionsMode: ColumnActionsMode.clickable,
-    onRender: (rec) => (
-      <Bar textWidth={70} value={rec.memory_max} capacity={capacity}>
-        {getValueFormat('bytes')(rec.memory_max, 1)}
-      </Bar>
-    ),
-  }
-}
-
-function txnStartTsColumn(
-  _rows?: { txn_start_ts?: number }[] // used for type check only
-): IColumn {
-  return {
-    name: commonColumnName('txn_start_ts'),
-    key: 'Txn_start_ts',
-    fieldName: 'txn_start_ts',
-    minWidth: 100,
-    maxWidth: 150,
-    onRender: (rec) => (
-      <Tooltip title={rec.txn_start_ts}>
-        <TextWrap>{rec.txn_start_ts}</TextWrap>
-      </Tooltip>
-    ),
+    onRender: (rec) => {
+      const tooltipContent = `
+Mean: ${getValueFormat('ns')(rec[avgFiledName], 1)}
+P90:  ${getValueFormat('ns')(rec[p90FiledName], 1)}
+Max:  ${getValueFormat('ns')(rec[maxFiledName], 1)}`
+      return (
+        <Tooltip title={<Pre>{tooltipContent.trim()}</Pre>}>
+          <Bar
+            textWidth={70}
+            value={rec[avgFiledName]}
+            max={rec[maxFiledName]}
+            min={rec[p90FiledName]}
+            capacity={capacity}
+          >
+            {getValueFormat('ns')(rec[avgFiledName], 1)}
+          </Bar>
+        </Tooltip>
+      )
+    },
   }
 }
 
 //////////////////////////////////////////
 
 export function slowQueryColumns(
-  rows: SlowqueryBase[],
+  rows: SlowquerySlowQuery[],
   showFullSQL?: boolean
 ): IColumn[] {
   return [
     sqlColumn(rows, showFullSQL),
-    digestColumn(rows),
-    instanceColumn(rows),
-    dbColumn(rows),
-    timestampColumn(rows),
-    queryTimeColumn(rows),
-    parseTimeColumn(rows),
-    compileTimeColumn(rows),
-    processTimeColumn(rows),
-    memoryColumn(rows),
-    txnStartTsColumn(rows),
+    textWithTooltipColumn(TRANS_KEY_PREFIX, 'Digest'),
```

Comment:
Maybe we can make some code refactoring to these columns. How about:

```js
const cf = new CommonColumnFactory(TRANS_KEY_PREFIX);
return [
  cf.textWithTooltip('digest'),
  ...
  cf.bar.single('query_time', 's', rows),
  cf.bar.multiple('cop_wait', 's', rows, {
    displayTransKey: '...',
    bars: {
      max: ..
      min: ..
      avg: ..
    }
  })
]
```

Reply:
Great! will try it.